### PR TITLE
SIM-1711 ADD documentation for new rule to prevent injections

### PR DESCRIPTION
**Pull request by gregor-balint**

### CODE REVIEW

File: docs/rules/escape-user-inputs-in-sql-queries.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+[[toc]]
+
+## Properties
+
+<RuleProperties />
+
+## Description
+
+This rule prevents Injection by escaping user input before putting it in a SQL-query. It can be applied when an ORACLE-database management system is used.
```
Comment (arditymeri) [inline docs/rules/escape-user-inputs-in-sql-queries.md:26]: We can mention some more details here: e.g:
 * Which libraries are required,
 * What do we use for escaping the user input (Codec, ESAPI...)

File: docs/rules/escape-user-inputs-in-sql-queries.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+```java
+HttpServletRequest req = getRequest();
+Codec<Character> ORACLE_CODEC = new OracleCodec();
+String query = "SELECT first_name FROM employee WHERE department_id ='" +  ESAPI.encoder().encodeForSQL(ORACLE_CODEC, req.getParameter("departmentId")) + "' ORDER BY last_name";
+Statement statement = connection.createStatement();
+statement.execute(query);
+ResultSet resultSet = statement.getResultSet();
+```
+
+__Pre__
```
Comment (arditymeri) [inline docs/rules/escape-user-inputs-in-sql-queries.md:66]: I think the example above is enough to create an idea. We don't want to overwhelm the reader with examples.

File: docs/rules/escape-user-inputs-in-sql-queries.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+```java
+HttpServletRequest req = getRequest();
+Codec<Character> ORACLE_CODEC = new OracleCodec();
+String query = "SELECT first_name FROM employee WHERE department_id ='" + ESAPI.encoder().encodeForSQL(ORACLE_CODEC, req.getParameter("departmentId")) + "' ORDER BY last_name";
+Statement statement = connection.createStatement();
+ResultSet resultSet = statement.executeQuery(query);
+```
+
+__Pre__
+```java
```
Comment (arditymeri) [inline docs/rules/escape-user-inputs-in-sql-queries.md:105]: In this case, too. One example with executeQuery is enough in my opinion.

File: docs/rules/escape-user-inputs-in-sql-queries.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+Codec<Character> ORACLE_CODEC = new OracleCodec();
+String query = "SELECT first_name FROM employee WHERE";
+query += " id > '" + ESAPI.encoder().encodeForSQL(ORACLE_CODEC, req.getParameter("id")) + "'";
+query += " AND department_id ='" + ESAPI.encoder().encodeForSQL(ORACLE_CODEC, req.getParameter("departmentId")) + "'";
+query += " ORDER BY last_name";
+Statement statement = connection.createStatement();
+ResultSet resultSet = statement.executeQuery(query);
+```
+
+__Pre__
```
Comment (arditymeri) [inline docs/rules/escape-user-inputs-in-sql-queries.md:146]: Here too. One example with multiple concatenation statements would be enough.

File: docs/rules/escape-user-inputs-in-sql-queries.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
+---
+title: Escape User Inputs in SQL Queries
+ruleId: EscapeUserInputsInSQLQueries
+since: 3.17.0
+minJavaVersion: 1.1
+remediationCost: 10
```
Comment (arditymeri) [inline docs/rules/escape-user-inputs-in-sql-queries.md:6]: The remediation cost in the rule implementation is set to 5 minutes. This property needs to match with the remediation cost in the implementation.

File: docs/rules/escape-user-inputs-in-sql-queries.md (hunk 2 above)
Comment (gregor-balint) [reply docs/rules/escape-user-inputs-in-sql-queries.md:66]: OK in all examples I will only take the example with the invocation of ```getParameter```

**Pull request merged**